include/OpenSpaceToolkit/Astrodynamics/EventCondition.hpp

```diff
@@ -28,16 +29,20 @@ class EventCondition
     /// @brief                  Constructor
     ///
     /// @code
-    ///                         EventCondition eventCondition = {aName, aCriterion, anEvaluatro, aTarget};
+    ///                         EventCondition eventCondition = {aName, anEvaluator, aTarget, targetIsRelative};
     /// @endcode
     ///
     /// @param                  [in] aName A string representing the name of the Real Event Condition
-    /// @param                  [in] aCriterion An enum indicating the criterion used to determine if the Real Event
-    /// Condition is met
-    /// @param                  [in] anEvaluator A function evaluating a state and a time
+    /// @param                  [in] anEvaluator A function evaluating a state
     /// @param                  [in] aTarget A target value associated with the Real Event Condition
+    /// @param                  [in] targetIsRelative A boolean indicating whether or not the target is relative
 
-    EventCondition(const String& aName);
+    EventCondition(
+        const String& aName,
+        std::function<Real(const State&)> anEvaluator,
+        const Real& aTarget,
+        const bool& targetIsRelative = false
```

Comment:
```suggestion
        const TargetSource& targetFrom = TargetSource::AbsoluteValue
```

Re: conversation about making this a bool vs. enum. If there will likely only be two values, the bool an enum approaches are similar and the bool approach is easier to implement.

Looking at the interface, it's a little unclear IMO what relative means here or how to use it. E.g. what is it relative to? I think if I approached this with fresh eyes, I would assume that the way to implement an event condition that relative to a value of 600km altitude is to use a relative event condition. 

This can probably be clearer with some naming like `targetRelativeToSegmentStart`. Maybe something better than that... just something to make it clear that it's relative to the start of the segment rather that a value known a-priori.

I like the semantics of an enum here for that reason, even if we don't technically need it for 2 values. Seems a little easier to indicate that the condition will depend on the initial Segment state at runtime. Something like:
```
    enum TargetSource
    {
        AbsoluteValue,
        SegmentStart,
        SequenceStart
    };
```

The `SequenceStart` value is not _really_ needed because as you pointed out, we always know the Sequence start State at construction time. However, it could be nice UX to be able to target the Sequence start automatically without having to copy the initial State around. This is a minor thing IMO though

Reply:
Yeah +1 on Enum. I don't think it's worth sublcassing EventCondition into a RelativeEventCondition style. I'll play with the Enum and see what I get.